tools/modify-image

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env bash
+
+set -eu -o pipefail
+shopt -qs failglob
+
+# check for veritysetup, debugfs, lz4
+IMAGE_PATH=""
+IMAGE=""
+OUT_IMAGE=""
+ROOTFILE=""
+CABUNDLE=""
+declare -a CHANGED_ROOTS
+
+function buildJson() {
+
+  local -n arr=$1
+  JSON_STRING=$( jq -n \
+      --arg path "${arr[Path]}" \
+      --arg nh "${arr[NewHash]}" \
+      --arg oh "${arr[OldHash]}" \
+      '{Path: $path, NewHash: $nh, OldHash: $oh}' )
+
+  CHANGED_ROOTS+=(${JSON_STRING})
+}
+
+for opt in "$@"; do
+   optarg="$(expr "${opt}" : '[^=]*=\(.*\)')"
+   case "${opt}" in
+      --image=*) IMAGE_PATH="${optarg}" ;;
+      --image-out=*) OUT_IMAGE="${optarg}" ;;
+      --new-root=*) ROOTFILE="${optarg}" ;;
+      --ca-bundle=*) CABUNDLE="${optarg}" ;;
+   esac
+done
+
+echo -e "here are the vars: ${IMAGE_PATH} ${OUT_IMAGE} ${ROOTFILE} ${CABUNDLE}"
+
+DESTDIR=$(dirname ${IMAGE_PATH})
+WORKDIR="$(mktemp -d)"
+cleanup() {
+   [ -n "${WORKDIR}" ] && rm -rf "${WORKDIR}"
+}
+trap 'cleanup' EXIT
+
+cp ${IMAGE_PATH} ${WORKDIR}
+
+pushd "${WORKDIR}" >/dev/null
+
+IMAGE=$(basename "${IMAGE_PATH}")
+
+if [[ ${IMAGE} == *.lz4 ]] ; then
+echo "Extracting ${IMAGE}"
+  rm -f ${IMAGE%.*}
+  unlz4 ${IMAGE}
+  IMAGE=${IMAGE%.*}
+fi
+
+IMAGE_NAME=${IMAGE%.*}
+# step 2: extract the root filesystem
+echo -e "Starting extract of root for ${IMAGE}"
+ROOT_IMAGE="root.ext4"
+ROOT_START="$(sgdisk --print "${IMAGE}" | awk '/BOTTLEROCKET-ROOT-A/{ print $2 }')"
+echo ${ROOT_START}
+ROOT_END="$(sgdisk --print "${IMAGE}" | awk '/BOTTLEROCKET-ROOT-A/{ print $3 }')"
+echo ${ROOT_END}
+ROOT_SIZE="$(( ROOT_END - ROOT_START + 1))"
```

Comment:
It might help to associate units with these - sectors, bytes, whatever - since it's not immediately obvious what the fields we're cutting out of `sgdisk` output. If `sgdisk --info` lets us match lines by a prefix then that might help too.

Maybe a helper function that takes a partition name and makes a couple `sgdisk` calls to find it and populate the variables that the caller passes in.